sub-packages/bionemo-geneformer/src/bionemo/geneformer/scripts/geneformer_mlm_loss_eval.py

```diff
@@ -0,0 +1,314 @@
+            ),
+        )
+
+        with torch.no_grad():
+            dl_hf_iter = iter(dl_hf)
+            dl_nv_iter = iter(dl_nv)
+            loss_hf = 0.0
+            n_hf = 0
+            loss_nv = 0.0
+            n_nv = 0
+            nv_device = geneformer_nv_inferer.module.embedding.position_embeddings.weight.device
+            hf_device = hf_model.device
+            for _ in trange(len(dl_hf)):
+                batch_hf = {k: v.to(hf_device) for k, v in next(dl_hf_iter).items()}
+                batch_nv = {k: v.to(nv_device) for k, v in next(dl_nv_iter).items()}
+                logits_hf = hf_model(batch_hf["text"].long(), batch_hf["attention_mask"])
+                loss_hf += (
+                    torch.nn.functional.cross_entropy(
+                        logits_hf[batch_hf["loss_mask"]],
+                        batch_hf["labels"][batch_hf["loss_mask"]],
+                        reduction="sum",
+                    )
+                    .cpu()
+                    .sum()
+                    .item()
+                )
+                n_hf += batch_hf["loss_mask"].sum().cpu().item()
+
+                logits_nv = (
+                    geneformer_nv_inferer(batch_nv["text"], batch_nv["attention_mask"])["token_logits"]
+                    .transpose(0, 1)
+                    .contiguous()
+                )
+                loss_nv += (
+                    torch.nn.functional.cross_entropy(
+                        logits_nv[batch_nv["loss_mask"]][..., :ori_nv_vocab_size],
+                        batch_nv["labels"][batch_nv["loss_mask"]],
+                        reduction="sum",
+                    )
+                    .cpu()
+                    .sum()
+                    .item()
+                )
+                n_nv += batch_nv["loss_mask"].sum().cpu().item()
+        print(f"NV mean loss: {loss_nv / n_nv}")
+        print(f"HF mean loss: {loss_hf / n_hf}")
+
+
+def entrypoint():
+    """Main entry point for running the evaluation."""
+    parser = argparse.ArgumentParser(description="MLM Performance vs HF Script")
+    parser.add_argument(
+        "--model-path",
+        type=Path,
+        help="Path to nvidia geneformer model checkpoint (unless you want random weights)",
+        required=False,
+        default=None,
+    )
+    parser.add_argument(
+        "--hf-token-dictionary-path",
+        type=Path,
+        help="Path to token dictionary file. "
+        "Eg `wget https://huggingface.co/ctheodoris/Geneformer/resolve/main/geneformer/token_dictionary_gc95M.pkl`"
+        "then provide the path to the downloaded file.",
+        required=True,
+    )
+    parser.add_argument(
+        "--hf-medians-dictionary-path",
+        type=Path,
+        help="Path to token dictionary file. "
+        "Eg `wget https://huggingface.co/ctheodoris/Geneformer/resolve/main/geneformer/gene_median_dictionary_gc95M.pkl` "
+        "then provide the path to the downloaded file.",
+        required=True,
+    )
```

Comment:
Any reason you couldn't just use `pooch` directly and point it to these files? You could save them to `BIONEMO_CACHE_DIR` and it should be easier to put a test around this

Reply:
I made an issue for this #401